### ISSUE TITLE
0.7 ignores renamed flavour output name

### ISSUE THREAD

**Issue by Bananeweizen** [KeepSafe/dexcount-gradle-plugin#181]
We rename the output file in one flavour, see [lines 202 and following of this build script](https://github.com/cgeo/cgeo/blob/6a8e90292043d5d1bb2b8703aa0c10f1c7e677d2/main/build.gradle#L202). This worked well up to 0.6.4. Since 0.7.0, the dexcount invocation looks for the original (non renamed) output file, thereby breaking the build:

```
* What went wrong:
A problem was found with the configuration of task ':main:countBasicDebugDexMethods'.
> File '/home/jenkins/slave/workspace/cgeo continuous integration/main/build/outputs/apk/cgeo-basic-debug.apk' specified for property 'inputFile' does not exist.
```

**Comment by benjamin-bader** [comment]
Good to know - thanks for the report!

**Comment by vanniktech** [comment]
You shouldn't do it like that. There's an archieveBasename property which you can use.

**Comment by benjamin-bader** [comment]
That's true, and the linked script will fail hard on upgrade to Android Studio 3, but in the meantime this is something we can likely fix very easily.

Be that as it may, `archiveBasename` (or whatever it's called) is definitely the right way forward.

**Comment by jp1017** [comment]
same issue of 0.7.0 . :sob:

**Comment by Bananeweizen** [comment]
Can I really use archivesBasename to modify the output of a single build type (and not for all build types together)?

And why do you think that renaming the output will no longer work at all in android-gradle 3.0? Reading https://developer.android.com/studio/preview/features/new-android-plugin-migration.html#variant_api I would expect that our build script just needs to use all() instead of each(), since the given example is very similar to what we do. Or did I miss something here?

I will experiment a bit with the given example code and the 3.0 plugin, to maybe find a workaround on our side.

**Comment by benjamin-bader** [comment]
> And why do you think that renaming the output will no longer work at all in android-gradle 3.0? Reading https://developer.android.com/studio/preview/features/new-android-plugin-migration.html#variant_api I would expect that our build script just needs to use all() instead of each(), since the given example is very similar to what we do. Or did I miss something here?

Because your script invokes `output.outputFile`, which (at least in alpha1 and alpha3) just throws a runtime exception.  `outputFileName` is a separate property, though its name makes that fact easy to overlook.

I think I must have misunderstood your use case; I haven't ever tried that with `archiveBasename`.  `outputFileName`, from the link you posted, seems like it would work.

**Comment by benjamin-bader** [comment]
FYI, I've just released version 0.7.1, which fixes this issue.

**Comment by jp1017** [comment]
@benjamin-bader great job, 0.7.1 works perfect. :+1: :smile:
